src/main/windows/windowManager.js

```diff
@@ -178,12 +178,56 @@ export function flashFrame(flash) {
   }
 }
 
-export function setOverlayIcon(overlayDataURL, description) {
+function drawBadge(text, small) {
+  const scale = 2; // should rely display dpi
+  const size = (small ? 20 : 16) * scale;
+  const canvas = document.createElement('canvas');
+  canvas.setAttribute('width', size);
+  canvas.setAttribute('height', size);
+  const ctx = canvas.getContext('2d');
+
+  // circle
+  ctx.fillStyle = '#FF1744'; // Material Red A400
+  ctx.beginPath();
+  ctx.arc(size / 2, size / 2, size / 2, 0, Math.PI * 2);
+  ctx.fill();
+
+  // text
+  ctx.fillStyle = '#ffffff';
+  ctx.textAlign = 'center';
+  ctx.textBaseline = 'middle';
+  ctx.font = (11 * scale) + 'px sans-serif';
+  ctx.fillText(text, size / 2, size / 2, size);
+
+  return canvas.toDataURL();
+}
+
+function createDataURL(text, small) {
+  const win = status.mainWindow;
+  if (!win) {
+    return null;
+  }
+
+  // since we don't have a document/canvas object in the main process, we use the webcontents from the window to draw.
+  const code = `
+    window.drawBadge = ${drawBadge};
+    window.drawBadge(${text}, ${small});
```

Comment:
How do the variables get expanded here? Shouldn't they be wrapped in quotes or `JSON.stringify`?

Reply:
the `small` one wouldn't be a problem as long as it is a boolean or undefined.
the text one should be surrounded, yes, but since my tests were involving numbers it didn't matter as it was used as an integer. but I'll fix those to prevent any errors